3ds/source/gui/overlay/PkmItemOverlay.cpp

```diff
@@ -70,15 +70,18 @@ PkmItemOverlay::PkmItemOverlay(ReplaceableScreen& screen, std::shared_ptr<PKX> p
 {
     instructions.addBox(false, 75, 30, 170, 23, COLOR_GREY, i18n::localize("SEARCH"), COLOR_WHITE);
     const std::vector<std::string>& rawItems = i18n::rawItems(Configuration::getInstance().language());
-    for (int i = 1; i <= TitleLoader::save->maxItem(); i++)
+    const std::set<int>& availableItems      = TitleLoader::save->availableItems();
+    for (auto i = availableItems.begin(); i != availableItems.end(); i++)
     {
-        if (rawItems[i].find("\uFF1F\uFF1F\uFF1F") != std::string::npos || rawItems[i].find("???") != std::string::npos)
+        if (*i == 0)
             continue;
-        else if (i >= 807 && i <= 835)
+        else if (rawItems[*i].find("\uFF1F\uFF1F\uFF1F") != std::string::npos || rawItems[*i].find("???") != std::string::npos)
```

Comment:
Could there be a possible out of bounds here?

Reply:
On a possible out of bounds: Sure, it's possible. However, that would only happen in the case that either A) the save's availableItems is configured incorrectly or B) we don't have strings for some moves

Reply:
I did combine the ifs, though